### PR TITLE
i18n

### PR DESCRIPTION
**Pull request by jgradim**
- Add i18next and react-i18next
 - Add recompose

Note: Still missing a lot of translations, but the groundwork is in place now

### CODE REVIEW

File: app/components/elements/Login.jsx [hunk 1]
```diff
@@ -1,29 +1,27 @@
...
 import { Link } from "react-router-dom";
+import { compose } from "recompose";
+import { translate } from "react-i18next";
 
 import { postLogin } from "lib/api";
 import { errorTransform } from "lib/errorTransform";
 import FormError from "components/elements/FormError";
 
-import t from "translations/pt.yml";
+// import t from "translations/pt.yml";
```
Comment (hugopeixoto) [inline app/components/elements/Login.jsx]: Remove this completely, please

File: app/components/elements/Login.jsx [hunk 2]
```diff
@@ -1,29 +1,27 @@
...
-      fields: {
-        email: "",
-        password: ""
-      },
-      errors: {}
-    };
+  state = {
+    fields: {
+      email: "",
+      password: ""
```
Comment (hugopeixoto) [inline app/components/elements/Login.jsx]: Missing dangling comma. Should we configure eslint to force it?

File: app/components/elements/Login.jsx [hunk 3]
```diff
@@ -1,29 +1,27 @@
...
+    },
+    errors: {}
```
Comment (hugopeixoto) [inline app/components/elements/Login.jsx]: Dangling comma is missing

File: app/components/pages/Home.jsx [hunk 4]
```diff
@@ -25,10 +26,12 @@ class Home extends React.Component {
   }
 
   render() {
+    const { t } = this.props;
+
     return (
       <div className="posts">
         <section className="post">
-          <h1 className="post-title">Programa de Mentorados de Engenharia Informática - FEUP</h1>
+          <h1 className="post-title">{t("home:title")}</h1>
```
Comment (hugopeixoto) [inline app/components/pages/Home.jsx]: Why is there a `home:` prefix here?

File: app/components/pages/Home.jsx [hunk 5]
```diff
@@ -74,17 +77,14 @@ class Home extends React.Component {
...
         <section>
-          <h2>Contacto</h2>
-
-          <p>
-            Se quiserdes contactar as pessoas responsaveis pela manutencao da
-            plataforma, com queixas, sugestoes, ou qualquer cena, mandai email
-            para <a href={`mailto:${email}`}>{email}</a>.
-          </p>
+          <h2>{t("contact.title")}</h2>
+          <p dangerouslySetInnerHTML={{ __html: t("contact.content", { email })}} />
```
Comment (hugopeixoto) [inline app/components/pages/Home.jsx:81]: I'm fine with letting this go for now, but eventually I'd like to find a solution for this.

**Review by hugopeixoto — COMMENTED** [review]
Most of the review notes are small style fixes.
The only major thing is the `dangerouslysetinnerhtml`shenanigans, which we can address later. Feel free to merge this after fixing the small style issues.

File: app/components/pages/Home.jsx (hunk 5 above)
Comment (jgradim) [reply app/components/pages/Home.jsx:81]: I think this is actually the cleanest way to achieve this. We should only use for values we define ourselves, but the other solutions provided by the `react-i18next` (`<Trans>`, `<Interpolate />`) are completely obtuse: https://react.i18next.com/components/trans-component.html

File: app/components/elements/Login.jsx (hunk 2 above)
Comment (jgradim) [reply app/components/elements/Login.jsx]: We definitely should

File: app/components/elements/Login.jsx (hunk 1 above)
Comment (jgradim) [reply app/components/elements/Login.jsx]: 🤦‍♂️

File: app/components/pages/Home.jsx (hunk 4 above)
Comment (jgradim) [reply app/components/pages/Home.jsx]: Holdover from when I was trying stuff out. Not necessary as we import the `home` namespace in the `translate([ "home" ])` HOC. However, if, we had imported multiple namespaces and not defined a default, we'd need to access translation strings like this. Removing namespace.

**Pull request merged**